### PR TITLE
Launchpad and Inscription api updates

### PR DESCRIPTION
**Pull request by T1-O**
#### APIs Added

- `Marketplace`

  - getListing - Updated request and response object.

- `Inscription`

  - createSpecialSats - Added method to create special sats.
  - getPrice - Updated the request and response object.
  - createOrder - Updated the request and response object.
  - createCollection - Updated the request and response object.
  - createCollectionOrder - Updated the request and response object.
  - createTextOrder - Updated the request and response object.

- `Launchpad`

  - constructor - Added method constructor for initialization of launchpad instance
  - createMarketPlace - Added method for creating a launchpad marketplace
  - createLaunchpad - Added method for creating a launchpad listing
  - getLaunchpadPSBT - Added method for retrieving the launchpad PSBT to sign transaction
  - saveLaunchpad - Added method for updating the signed PSBT on the launchpad
  - getLaunchpadListing - Added method for retrieving the active launchpad listing
  - getLaunchpadListing - Updated request and response object.
  - getAllocation - Added method for buyers to check allowed and claimed inscription for launchpad phases
  - confirmPaddingOutputs - Added method to check if the padding outputs are available in the buyer account
  - setupPaddingOutputs - Added method to set up if the padding outputs are not available in the buyer account
  - createLaunchpadOffer - Added method for buyers to request the ordinal from launchpad phase to buy
  - submitLaunchpadOffer - Added method to update the signed PSBT and broadcast the buyer transaction

- Ordinalsbot  

  - Added `Launchpad` instance in the Ordinalsbot class
  - Updated Usage of Ordinalsbot.
  
    ```
    import { Ordinalsbot } from 'ordinalsbot'

    // If no parameter is given, the default environment is 'live'
    const ordinalsbotObj = new Ordinalsbot(API_KEY, 'dev')
    const marketPlace = ordinalsbotObj.MarketPlace
    const inscription = ordinalsbotObj.Inscription

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,14 +24,18 @@ Install the package with:
...
-let satscanner = new Satscanner("API_KEY", "dev");
-let satextractor = new Satextractor("API_KEY", "dev");
-let mempool = new Mempool("API_KEY", "dev");
+const ordinalsbotObj = new Ordinalsbot(API_KEY, 'dev')
+const marketPlace = ordinalsbotObj.MarketPlace()
+const inscription = ordinalsbotObj.Inscription()
+const launchpad = ordinalsbotObj.Launchpad()
+const mempool = ordinalsbotObj.Mempool()
+const satextractor = ordinalsbotObj.Satextractor()
+const satscanner = ordinalsbotObj.Satscanner()
```
Comment (ordinariusprof) [inline README.md:37]: will this change break currently running apps? 
if so, we need to bump a major version right?

File: src/client.ts [hunk 2]
```diff
@@ -14,144 +14,217 @@ import {
...
 
     let config = {
-      method: "post",
+      method: 'post',
       maxBodyLength: Infinity,
-      url: this.instanceV1.getUri() + "/collectioncreate",
+      url: this.instanceV1.getUri() + '/collectioncreate',
       headers: {
-        "Content-Type": "application/x-www-form-urlencoded",
+        'Content-Type': 'application/x-www-form-urlencoded',
```
Comment (ordinariusprof) [inline src/client.ts:157]: this doesn't look right. we accept json in the backend.

File: src/client.ts [hunk 3]
```diff
@@ -14,144 +14,217 @@ import {
...
+    return this.instanceV1.post(`/textorder`, order)
   }
 
+  /**
+   * Retrieves inventory information.
+   * @returns {Promise<InscriptionInventoryResponse[]>} A promise resolving with the inventory information.
+   */
   async getInventory(): Promise<InscriptionInventoryResponse[]> {
-    return this.instanceV1.get(`/inventory`);
+    return this.instanceV1.get(`/inventory`)
```
Comment (ordinariusprof) [inline src/client.ts:191]: I think we should agree on the same format. we should use `;` after each line so let's do it in this repo too. I don't know why this had to be deleted.
Can you add a common eslint file so we're not changing format with each PR. it's messy. we should stick to same formatting rules.

**Review by ordinariusprof — APPROVED** [review]
just have a few questions on backwards compatibility and formatting. otherwise lgtm.

File: README.md (hunk 1 above)
Comment (T1-O) [reply README.md:37]: Yeah, I need to sort out the formatting

File: src/client.ts (hunk 2 above)
Comment (T1-O) [reply src/client.ts:157]: We already had  "Content-Type": "application/x-www-form-urlencoded" there is no change here.

File: README.md (hunk 1 above)
Comment (ordinariusprof) [reply README.md:37]: so what's the solution? will you change it to make it backwards compatible or are you bumping a major version?

**Pull request merged**